src/core/execute.c

```diff
@@ -346,6 +346,25 @@ bool exec_needs_mount_namespace(
         return false;
 }
 
+const char* exec_get_effective_notify_socket_path(const ExecContext *context, const ExecParameters *params, bool needs_sandboxing) {
```

Comment:
hmm, this is not very "effective". you return the private notify socket path only here, never the host one, hence the "effective" one is sometimes this, sometimes something else, and I think "effective" is hence a bit of a misnomer.

Maybe call this "exec_get_private_notify_socket_path()"?